### ISSUE TITLE
RFC 3433 support

### ISSUE THREAD

**Issue by RichiH** [prometheus/snmp_exporter#278]
[RFC 3433](https://tools.ietf.org/html/rfc3433) and its children (e.g. [CISCO-ENTITY-SENSOR-MIB](https://github.com/simonjj/SnmpMibs/blob/master/CISCO-ENTITY-SENSOR-MIB.mib) ) shoehorn floats into integers as such:

`entPhySensorValue * 10^-(entPhySensorPrecision) * si_prefix_lookup(entPhySensorScale)`

where `si_prefix_lookup()` would use

````
        yocto(1),   -- 10^-24
        zepto(2),   -- 10^-21
        atto(3),    -- 10^-18
        femto(4),   -- 10^-15
        pico(5),    -- 10^-12
        nano(6),    -- 10^-9
        micro(7),   -- 10^-6
        milli(8),   -- 10^-3
        units(9),   -- 10^0
        kilo(10),   -- 10^3
        mega(11),   -- 10^6
        giga(12),   -- 10^9
        tera(13),   -- 10^12
        exa(14),    -- 10^15
        peta(15),   -- 10^18
        zetta(16),  -- 10^21
        yotta(17)   -- 10^24
````
for lookup.

The obvious issue is that you are now stuck with hard-coding all possible combinations of `entPhySensorType`, `entPhySensorPrecision`, and `entPhySensorScale` in Grafana for all dashboards, or you write out the currently-seen combinations and alert on seeing new ones.

Ignoring that ASR920 forces you to substring match to find Volt and Ampere if you want to know how many Watt your devices is using, we would now need to take all combinations of both label sets and their respective permutations into account in order to display or alert on power consumption. This is between painful and overkill.

Long story short: As sensor MIBs are somewhat well-defined, it would be useful to encode this knowledge somewhere into snmp_exporter or generator.

If this is out of scope, I would even somewhat agree, given how broken this system is; though I would argue that it's, sadly, a valid long-term goal.

**Comment by brian-brazil** [comment]
This is something that'll have to be done in PromQL, as doing math across objects is going a bit too far. PromQL can handle the math required reasonably easily.

**Comment by RichiH** [comment]
Can you share a snippet how to do this in promql? I would happily work this
into our documentation.

**Comment by brian-brazil** [comment]
Offhand it's `entPhySensorValue * 10^-(entPhySensorPrecision) *  10^((entPhySensorScale-9)*3)`
